src/viewer.js

```diff
@@ -876,6 +876,201 @@ $.extend( $.Viewer.prototype, $.EventSource.prototype, $.ControlDock.prototype,
         return this;
     },
 
+    /**
+     * Open tiled images into the viewer, closing any others, and return a Promise
+     * that resolves when all images are loaded successfully or rejects on failure.
+     * @function
+     * @param {Array|String|Object|Function} tileSources - This can be a TiledImage
+     * specifier, a TileSource specifier, or an array of either. A TiledImage specifier
+     * is the same as the options parameter for {@link OpenSeadragon.Viewer#addTiledImage},
+     * except for the index property; images are added in sequence.
+     * A TileSource specifier is anything you could pass as the tileSource property
+     * of the options parameter for {@link OpenSeadragon.Viewer#addTiledImage}.
+     * @param {Number} initialPage - If sequenceMode is true, display this page initially
+     * for the given tileSources. If specified, will overwrite the Viewer's existing initialPage property.
+     * @returns {Promise} A promise that resolves when all tile sources are opened successfully,
+     *          or rejects with an error if any fail to load.
+     * @fires OpenSeadragon.Viewer.event:open
+     * @fires OpenSeadragon.Viewer.event:open-failed
+     */
+    openAsync: function (tileSources, initialPage) {
+        const _this = this;
+
```

Comment:
You should probably test if supportsAsync is true, and throw error otherwise. Or ensure that if $.Promise is stubbed without promise support, it actually works.

Reply:
I'm pretty sure we don't support browsers that don't have Promise built-in anymore, so I don't think that's a concern.

Reply:
Oh wow, I had forgotten about that.

I'd be happy to remove supportsAsync; I don't think we should be introducing non-standard behavior into standard components like Promise. What do you think?

That seems like a conversation for outside this PR, though. In terms of this PR, maybe the solution is to use non-OSD Promise here? It's not like we can make loading images be synchronous anyway.